src/main/java/org/opensearch/knn/indices/Model.java

```diff
@@ -13,23 +13,30 @@
 
 import org.apache.commons.lang.builder.EqualsBuilder;
 import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.opensearch.common.Nullable;
 
 import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
 
 public class Model {
 
-    final private ModelMetadata modelMetadata;
-    final private byte[] modelBlob;
+    private ModelMetadata modelMetadata;
+    private AtomicReference<byte[]> modelBlob;
 
     /**
      * Constructor
      *
      * @param modelMetadata metadata about the model
-     * @param modelBlob binary representation of model template index
+     * @param modelBlob binary representation of model template index. Can be null if model is not yet in CREATED state.
      */
-    public Model(ModelMetadata modelMetadata, byte[] modelBlob) {
+    public Model(ModelMetadata modelMetadata, @Nullable byte[] modelBlob) {
         this.modelMetadata = Objects.requireNonNull(modelMetadata, "modelMetadata must not be null");
-        this.modelBlob = Objects.requireNonNull(modelBlob, "modelBlob must not be null");
+
+        if (ModelState.CREATED.equals(this.modelMetadata.getState()) && modelBlob == null) {
+            throw new IllegalArgumentException("Model blob cannot be null when model metadata says model is created");
```

Comment:
Will user understands what is model blob/ model meta data?

Reply:
Right, we may have had this discussion before. Basically a model is made of a binary model and some metadata about the model. model_blob refers to the binary. I am not sure if there is a better name. Do you have any recs?

Reply:
As long as we document what this error message means / how to resolve it, it should be fine.

Reply:
Agree with Vijay. We can put message something like "We could not find model associated to this id xyz"?

Reply:
Sure, will update error message.